cli/azd/pkg/ext/hooks_runner.go

```diff
@@ -0,0 +1,171 @@
+package ext
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/bash"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/powershell"
+)
+
+// Hooks enable support to invoke integration scripts before & after commands
+// Scripts can be invoked at the project or service level or
+type HooksRunner struct {
+	hooksManager  *HooksManager
+	commandRunner exec.CommandRunner
+	console       input.Console
+	cwd           string
+	scripts       map[string]*ScriptConfig
```

Comment:
Just curious, is there a reason you used `*ScriptConfig` here instead of just `ScriptConfig`?  Would we ever expect a nil ScriptConfig in this map?

Reply:
Just looking to reduce copying values where we could could reference by pointer.